### PR TITLE
Interactivity styling

### PR DESCRIPTION
**Pull request by davidmanzanares**
Add https://github.com/CartoDB/renderer-prototype/issues/202

- [x] PoC
- [x] Generalized API
- [x] Reset with fadeOut
- [x] Fix broken tests
- [x] Add docs
- [x] String API 

Add tests? (it would be a combination of user and render tests... quite complex)

## Implementation idea

On feature color change:
- Backup previous color expression
- Create a new color expression that conditionally resolves to the new color if the feature is the selected one.
- Replace the root style color expression with the new one.

On feature color normalization:
- Restore previous color expression

## Pros
- Very easy to implement
- Public API semantics are easy

## Cons
- Accumulating per-feature styles is a nice way to lose all performance. Don't expect to color 200 features differently this way, and achieve good performance.

### CODE REVIEW

File: examples/interactivity/feature-click-style.html [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        const source = new carto.source.Dataset('ne_adm0_europe');
+        const style = new carto.Style(`
+      color: ramp($admin, bold)
+    `);
+        const layer = new carto.Layer('layer', source, style);
+
+        layer.on('loaded', () => {
+            updateTitle('Click on the countries');
+
+            const interactivity = new carto.Interactivity([layer]);
```
Comment (Jesus89) [inline examples/interactivity/feature-click-style.html]: this could be moved outside the on('loaded'

Comment (davidmanzanares) [reply examples/interactivity/feature-click-style.html]: Totally

File: test/integration/user/api/interactivity.test.js [hunk 2]
```diff
@@ -67,6 +67,10 @@ describe('Interactivity', () => {
             it('should fire a featureClick event with a features list containing the clicked feature', done => {
                 interactivity = new carto.Interactivity(layer);
                 interactivity.on('featureClick', event => {
+                    expect(event.features[0].style.color.blendTo).toBeDefined();
+                    expect(event.features[0].style.color.reset).toBeDefined();
+                    expect(event.features[0].style.reset).toBeDefined();
+                    delete event.features[0].style;
```
Comment (Jesus89) [inline test/integration/user/api/interactivity.test.js]: Please use this instead:
```js
expect(event.features[0]).toEqual({ id: 0, layerId: 'layer', properties: { cartodb_id: 0, style: jasmine.any(Object) } });
```

File: test/integration/user/api/interactivity.test.js [hunk 3]
```diff
@@ -103,6 +107,8 @@ describe('Interactivity', () => {
                 layer = new carto.Layer('layer', source, style);
                 interactivity = new carto.Interactivity(layer);
                 interactivity.on('featureClick', event => {
+                    delete event.features[0].style;
+                    delete event.features[1].style;
                     expect(event.features[0]).toEqual({ id: 0, layerId: 'layer', properties: { cartodb_id: 0 } });
```
Comment (Jesus89) [inline test/integration/user/api/interactivity.test.js]: [^ comment](https://github.com/CartoDB/renderer-prototype/pull/232/files#r178863960)

File: test/integration/user/api/interactivity.test.js [hunk 4]
```diff
@@ -103,6 +107,8 @@ describe('Interactivity', () => {
                 layer = new carto.Layer('layer', source, style);
                 interactivity = new carto.Interactivity(layer);
                 interactivity.on('featureClick', event => {
+                    delete event.features[0].style;
+                    delete event.features[1].style;
```
Comment (Jesus89) [inline test/integration/user/api/interactivity.test.js]: [^ comment](https://github.com/CartoDB/renderer-prototype/pull/232/files#r178863960)

File: test/integration/user/api/interactivity.test.js (hunk 2 above)
Comment (davidmanzanares) [reply test/integration/user/api/interactivity.test.js]: Yes... That's better I asked off-channel about exactly this before doing this ñapa :crying_cat_face:

File: src/core/renderLayer.js [hunk 5]
```diff
@@ -41,7 +44,66 @@ export default class RenderLayer {
...
+                        return;
+                    }
+                    const blendExpr = blend(
+                        newExpression,
+                        this.style._styleSpec[styleProperty],
+                        blend(1, notEquals(property('cartodb_id'), feature.id), animate(duration))
+                    );
+                    this.trackFeatureStyle(feature.id, styleProperty, blendExpr);
+                    this.style.replaceChild(
+                        this.style.getColor(),
```
Comment (Jesus89) [inline src/core/renderLayer.js]: getColor()? What about the rest of properties?

Comment (davidmanzanares) [reply src/core/renderLayer.js]: LOL, really good catch

File: test/integration/user/api/interactivity.test.js [hunk 6]
```diff
@@ -99,7 +99,6 @@ describe('Interactivity', () => {
                 });
                 layer.addTo(map);
             });
-        });
```
Comment (Jesus89) [inline test/integration/user/api/interactivity.test.js]: Ups

**Pull request merged**